api/main.py

```diff
@@ -3,18 +3,54 @@
 from fastapi.exceptions import RequestValidationError
 from fastapi.responses import JSONResponse
 # from controller import Controller
+from object_store import ObjectStore
+import boto3
 
 import pickle
-import json
+import json, os
+import logging
 
 import numpy
 
+COMPLETED_BUCKET = os.getenv("COMPLETED_BUCKET")
+PENDING_BUCKET = os.getenv("PENDING_BUCKET")
+s3 = boto3.client("s3")
+# Initialize ObjectStore
+try:
+    ob = ObjectStore()
+except Exception as ex:
+    logging.error("Error is -", ex)
+
+
 app = FastAPI()
 
 @app.get("/")
 async def homepage(request: Request):
     return("Welcome to Async Job Framework")
 
+
+@app.get("/getResult/{job_id}")
+async def getResult(job_id: str):
+    """Get the job id and try fetching the result."""
+    # TODO add error response for user, add logic to check in pending bucket if not found in completed.
+    job_id_extension = str(job_id).strip()+".json"
+    try:
+        result = ob.get_object(job_id_extension,COMPLETED_BUCKET)
+        #check if result is available or not in completed bucket and if not available check for the job in pending bucket
+        if result is None:
+            for key in s3.list_objects(Bucket='quantumpending')['Contents']:
```

Comment:
@varshithauppula96 why are we using s3 here instead of the object store method? Also, why is `quantumpending` hardcoded?